lib/sipInterconnect.js

```diff
@@ -106,50 +100,65 @@ var api = function(config, method, path, body, callback) {
-  if(typeof options == 'function') {
+  if (typeof options == 'function') {
     callback = options;
     options = {};
   }
-  if(typeof callback != 'function') {
+  if (typeof callback !== 'function') {
     throw(new errors.ArgumentError('No callback given to dial'));
   }
-  if(sessionId == null || sessionId.length === 0) {
+  if (sessionId == null || sessionId.length === 0) {
     callback(new errors.ArgumentError('No session ID given'));
     return;
   }
+  if (token == null || token.length === 0) {
+    callback(new errors.ArgumentError('No token given'));
+    return;
+  }
+  if (sipUri == null || sipUri.length === 0) {
+    callback(new errors.ArgumentError('No SIP URI given'));
+    return;
+  }
   var body = {
-	sessionId: sessionId,
-	token: token,
-	sip: {
-		uri: sipUri
-	}
+    sessionId: sessionId,
+    token: token,
+    sip: {
+      uri: sipUri
+    }
   };
   if (_.isObject(options.headers) && !_.isArray(options.headers)) {
-  	body.sip.headers = options.headers;
+    body.sip.headers = options.headers;
   };
   if (_.isObject(options.auth) && !_.isArray(options.auth)) {
-  	body.sip.auth = options.auth;
+    body.sip.auth = options.auth;
   };
   if (!!options.secure) {
-  	body.sip.secure = options.secure;
+    body.sip.secure = !!options.secure;
   }
-  api(config, 'POST', '/call', body,
-  function(err, response, body) {
-  	console.log([response.statusCode, body]);
-    if(err) {
+  api(config, 'POST', '/call', body, function(err, response, body) {
+    if (err) {
       callback(err);
-    } else if(response.statusCode != 200) {
-      if(response && response.statusCode == 404) {
-        callback(new errors.SipError('Session not found'));
-      } else if(response && response.statusCode == 403) {
+    } else if (response.statusCode !== 200) {
+      if (response && response.statusCode == 400) {
+        if (body.message.indexOf('Bad SIP URI format') > -1) {
+          callback(new errors.SipError('Bad SIP URI format, only uris with format ' +
+            'sip:username@domain and optional transport paramater are supported.'));
+        } else {
+          callback(new errors.SipError('Session not found'));
+        }
+      } else if (response && response.statusCode == 403) {
         callback(new errors.AuthError('Invalid API key or secret'));
+      } else if (response && response.statusCode == 409) {
+        callback(new errors.SipError('Only Routed Sessions are allowed to initiate SIP '
+          + 'Calls.'));
       } else {
         callback(new errors.RequestError('Unexpected response from OpenTok'));
-      }    
-    } else if(!body.hasOwnProperty('streamId')) {
-      callback(new errors.RequestError('Unexpected response from OpenTok'));    
+      }
     } else {
       callback(null, new SipInterconnect(config, body));
     }
   });
 };
+
+//For testing suite with instanceof, do NOT use this
+exports._constructor_ = SipInterconnect;
```

Comment:
wat? perhaps change the exported symbol to SipInterconnect, and make `dial()` a method. i get that you're going to a more functional style, but if typeof comparisons are important to you (as it seems from your tests) then you really want to use the object oriented style to your advantage.